contracts/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@uma/core/contracts/oracle/interfaces/StoreInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OracleAncillaryInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/FinderInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleV2Interface.sol";
+
+import "@uma/core/contracts/common/implementation/AncillaryData.sol";
+import "@uma/core/contracts/common/implementation/AddressWhitelist.sol";
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+
+contract OptimisticArbitrator is Testable {
+    using SafeERC20 for IERC20;
+
+    struct Request {
+        address proposer; // Address of the proposer.
+        address disputer; // Address of the disputer.
+        IERC20 currency; // ERC20 token used to pay rewards and fees.
+        bool settled; // True if the request is settled.
+        int256 proposedPrice; // Price that the proposer submitted.
+        uint256 reward; // Amount of the currency to pay to the proposer on settlement.
+        uint256 finalFee; // Final fee to pay to the Store upon request to the DVM.
+        uint256 bond; // Bond that the proposer and disputer must pay on top of the final fee.
+        uint64 customLiveness; // Custom liveness value set by the requester.
+        uint64 expirationTime; // Time at which the request auto-settles without a dispute.
+    }
+
+    FinderInterface public finder;
+
+    bytes32 public priceIdentifier = "YES_OR_NO_QUERY";
+
+    uint256 public constant OO_ANCILLARY_DATA_LIMIT = 8139; // 8192 - 53
+
+    constructor(address _finderAddress, address _timerAddress) Testable(_timerAddress) {
+        finder = FinderInterface(_finderAddress);
+    }
+
+    mapping(bytes32 => Request) public requests;
+
+    function requestPrice(
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        IERC20 currency,
```

Comment:
one simplification would be to hardcode the currency? we always use a provided WETH collateral `currency`, potentially. Equally, we can hard code `customLiveness`? Alternatively, if we re-name the `customLiveness` to `liveness`? I'm trying to ideally get this function definition down to one line, so it does not need to wrap like this.